src/libpriv/rpmostree-passwd-util.c

```diff
@@ -384,9 +384,10 @@ rpmostree_check_passwd_groups (gboolean         passwd,
   if (!new_contents)
       return FALSE;
 
+  ignore_removed_ents = g_ptr_array_new ();
```

Comment:
WDYT of keeping this under the if statement so that we only allocate if we need to (as before), and instead move the `ignore_all_removed` to inside the if statement as well so that it stays with its default of `FALSE` otherwise?

Reply:
I do agree with moving `ignore_all_removed` inside the if statement, and having the allocation be done in the if condition makes sense.

The issue I see with having the allocation within the if statement is `ignore_removed_ents` will be passed as null elsewhere ([line 437](https://github.com/projectatomic/rpm-ostree/pull/1507/files#diff-dd7cf8a2b651ac4a3d12d0ea264bb29dR437)  in the passwd case).

This could be a workaround:
- have `(ignore_removed_ents != NULL && rpmostree_str_ptrarray_contains (ignore_removed_ents, odata->name))` on [line 437](https://github.com/projectatomic/rpm-ostree/pull/1507/files#diff-dd7cf8a2b651ac4a3d12d0ea264bb29dR437) to short circuit the && before accessing `ignore_removed_ents`

That way, there is a `!= NULL` check required before using `rpmostree_str_prtarray_contains`, but memory will be allocated only within the if statement. How does this sound?

Reply:
Sounds good! Added in the `NULL` check to treat it as an empty list.

Also added a `g_assert()` for `str` to handle `NULL` for the `const char *str` parameter as `g_str_equal()` does not expect `NULL`. My thinking is that something similar has been done [here](https://github.com/projectatomic/rpm-ostree/blob/0206f7a82fb867e24c66df768223182efd485780/src/libpriv/rpmostree-util.c#L1020).

`g_strcmp0()` could be used alternatively if we wanted to handle `str==NULL` gracefully. I did not go this route as we don't currently need to check the array for containing `NULL` pointers. However, using this function does not seem to cause any other problems (seeing it also used in `libpriv/rpmostree-core.c`), so it could be done.